### PR TITLE
Increase and enhance fuzzy testing

### PR DESCRIPTION
**Pull request by Xiaohanjlll**

### CODE REVIEW

File: unit_test/fuzzing/test_requester/test_spdm_requester_get_capabilities/get_capabilities.c [hunk 1]
```diff
@@ -10,6 +10,13 @@
 
 uint8_t temp_buf[LIBSPDM_RECEIVER_BUFFER_SIZE];
 
+spdm_version_number_t m_version[10] = {
```
Comment (Zhiqiang520) [inline unit_test/fuzzing/test_requester/test_spdm_requester_get_capabilities/get_capabilities.c]: `m_version[10]`
why select `10` here? why not select `4` or `20` and so on?

File: unit_test/fuzzing/test_requester/test_spdm_requester_get_capabilities/get_capabilities.c [hunk 2]
```diff
@@ -63,7 +70,14 @@ void libspdm_test_requester_get_capabilities(void **State)
                                             SPDM_VERSION_NUMBER_SHIFT_BIT;
     spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_VERSION;
 
-    libspdm_get_capabilities(spdm_context);
+    for (uint8_t index = 0; index < sizeof(m_version)/2; index++) {
```
Comment (Zhiqiang520) [inline unit_test/fuzzing/test_requester/test_spdm_requester_get_capabilities/get_capabilities.c]: `sizeof(m_version)/2`
why?
Maybe we change it as following to make it more clear?
`sizeof(m_version) / sizeof(m_version[0])`

File: unit_test/fuzzing/test_requester/test_spdm_requester_negotiate_algorithms/negotiate_algorithms.c [hunk 3]
```diff
@@ -161,6 +165,98 @@ void libspdm_test_requester_negotiate_algorithms_case4(void **State)
...
+        SPDM_GET_CAPABILITIES_REQUEST_FLAGS_MULTI_KEY_CAP_ONLY;
+    spdm_context->local_context.algorithm.other_params_support = SPDM_ALGORITHMS_MULTI_KEY_CONN;
+
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    libspdm_reset_message_a(spdm_context);
+
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 5: MULTI_KEY_CAP set 01*/
+    spdm_context->connection_info.capability.flags =
```
Comment (Zhiqiang520) [inline unit_test/fuzzing/test_requester/test_spdm_requester_negotiate_algorithms/negotiate_algorithms.c:248]: line 247 `/* Sub Case 5: MULTI_KEY_CAP set 01*/` same with line 235 `/* Sub Case 5: MULTI_KEY_CAP set 01*/`

File: unit_test/fuzzing/test_responder/test_spdm_responder_algorithms/algorithms.c [hunk 4]
```diff
@@ -310,7 +310,7 @@ void libspdm_test_responder_algorithms_case10(void **State)
                                     spdm_test_context->test_buffer, &response_size, response);
 }
 
-void libspdm_test_responder_algorithms_case11(void **State)
+void libspdm_test_responder_algorithms_case12(void **State)
 {
```
Comment (Zhiqiang520) [inline unit_test/fuzzing/test_responder/test_spdm_responder_algorithms/algorithms.c]: what is the difference between `libspdm_test_responder_algorithms_case11` and `libspdm_test_responder_algorithms_case12`?
why you rename here?
can you add some comment for the new test case to describe its test scenario?

File: unit_test/fuzzing/test_responder/test_spdm_responder_challenge_auth/challenge_auth.c [hunk 5]
```diff
@@ -306,6 +402,13 @@ void libspdm_run_test_harness(void *test_buffer, size_t test_buffer_size)
     libspdm_test_responder_challenge_case6(&State);
     libspdm_unit_test_group_teardown(&State);
 
+    libspdm_unit_test_group_setup(&State);
+    libspdm_test_responder_challenge_case7(&State);
+    libspdm_unit_test_group_teardown(&State);
+
+    libspdm_unit_test_group_setup(&State);
+    libspdm_test_responder_challenge_case8(&State);
+    libspdm_unit_test_group_teardown(&State);
```
Comment (Zhiqiang520) [inline unit_test/fuzzing/test_responder/test_spdm_responder_challenge_auth/challenge_auth.c:413]: can you add some comment for the new test case7 and case8 to describe its test scenario?

**Comment by Xiaohanjlll** [comment]
Old coverage
libspdm_req_challenge.c	69.0%
libspdm_req_get_capabilities.c	75.5%
libspdm_req_get_certificate.c	71.2%
libspdm_req_get_csr.c	75.9%
libspdm_req_get_digests.c	76.6%
libspdm_req_get_measurements.c	75.2%
libspdm_req_get_version.c	81.2%
libspdm_req_heartbeat.c	92.0%
libspdm_req_key_exchange.c	75.9%
libspdm_req_key_update.c	82.9%
libspdm_req_negotiate_algorithms.c	81.8%
libspdm_req_psk_exchange.c	76.6%
libspdm_req_psk_finish.c	89.0%
libspdm_req_set_certificate.c	77.1%

File: unit_test/fuzzing/test_requester/test_spdm_requester_negotiate_algorithms/negotiate_algorithms.c [hunk 6]
```diff
@@ -161,6 +165,98 @@ void libspdm_test_requester_negotiate_algorithms_case4(void **State)
...
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 4: MEL_CAP set 0, mel_specification set 0*/
+    spdm_context->connection_info.capability.flags = connection_capability_flags;
+    libspdm_reset_message_a(spdm_context);
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    spdm_context->local_context.algorithm.mel_spec = 0;
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 5: MULTI_KEY_CAP set 01*/
```
Comment (Zhiqiang520) [inline unit_test/fuzzing/test_requester/test_spdm_requester_negotiate_algorithms/negotiate_algorithms.c]: Change
`/* Sub Case 5: MULTI_KEY_CAP set 01*/`
To
`/* Sub Case 5: MULTI_KEY_CAP set 01b*/`

File: unit_test/fuzzing/test_requester/test_spdm_requester_negotiate_algorithms/negotiate_algorithms.c [hunk 7]
```diff
@@ -161,6 +165,98 @@ void libspdm_test_requester_negotiate_algorithms_case4(void **State)
...
+    spdm_context->local_context.capability.flags =
+        SPDM_GET_CAPABILITIES_REQUEST_FLAGS_MULTI_KEY_CAP_ONLY;
+    spdm_context->local_context.algorithm.other_params_support = SPDM_ALGORITHMS_MULTI_KEY_CONN;
+
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    libspdm_reset_message_a(spdm_context);
+
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 6: MULTI_KEY_CAP set 10*/
```
Comment (Zhiqiang520) [inline unit_test/fuzzing/test_requester/test_spdm_requester_negotiate_algorithms/negotiate_algorithms.c]: Change
`/* Sub Case 6: MULTI_KEY_CAP set 10*/`
To
`/* Sub Case 6: MULTI_KEY_CAP set 10b*/`

**Comment by Xiaohanjlll** [comment]
Enhance fuzzy testing by filtering and modifying seeds to improve the efficiency of fuzzy testing.

```
test_spdm_responder_algorithms/algorithms.c                 done
test_spdm_responder_capabilities/capabilities.c             done
test_spdm_responder_certificate/certificate.c               done
test_spdm_responder_challenge_auth/challenge_auth.c         done
test_spdm_responder_csr/csr.c                               done
test_spdm_responder_digests/digests.c                       done
test_spdm_responder_finish_rsp/finish_rsp.c                 done
test_spdm_responder_heartbeat_ack/heartbeat_ack.c           done
test_spdm_responder_key_exchange/key_exchange.c             done
test_spdm_responder_key_update/key_update.c                 done
test_spdm_responder_measurements/measurements.c             done
test_spdm_responder_psk_exchange_rsp/psk_exchange_rsp.c     done
test_spdm_responder_psk_finish_rsp/psk_finish_rsp.c         done
test_spdm_responder_set_certificate/set_certificate_rsp.c   done
test_spdm_responder_version/version.c                       done
```

**Comment by Xiaohanjlll** [comment]
Increase fuzzy testing, cover version 1.3, and improve coverage rate.    

 ```
libspdm_req_challenge.c	
libspdm_req_get_certificate.c
libspdm_req_set_certificate.c	
libspdm_req_get_measurements.c
libspdm_req_encap_challenge_auth.c
libspdm_req_encap_certificate.c
libspdm_req_encap_digests.c	
libspdm_req_negotiate_algorithms.c          81.8%->?        dnoe
libspdm_req_get_capabilities.c              75.5%->?        dnoe 
libspdm_rsp_capabilities.c                  91.6%->?        dnoe 
libspdm_rsp_algorithms.c                    61.6%->?        dnoe 
libspdm_rsp_challenge_auth.c
libspdm_rsp_csr.c
libspdm_rsp_digests.c
libspdm_rsp_encap_challenge.c
libspdm_rsp_encap_get_certificate.c	
libspdm_rsp_encap_get_digests.c	
libspdm_rsp_psk_finish.c	
libspdm_rsp_set_certificate.c
libspdm_rsp_certificate.c
libspdm_rsp_measurements.c
```

**Pull request closed without merging**